### PR TITLE
[LLVM][C API] Clearing initializer and personality by passing NULL.

### PR DESCRIPTION
**Pull request by maleadt**
This is similar to how the C++ API supports passing `nullptr` to `setPersonalityFn` or `setInitializer`.

### CODE REVIEW

File: llvm/lib/IR/Core.cpp [hunk 1]
```diff
@@ -1328,8 +1328,12 @@ LLVMValueRef LLVMMDNode(LLVMValueRef *Vals, unsigned Count) {
...
 }
 
-LLVMValueRef LLVMMetadataAsValue(LLVMContextRef C, LLVMMetadataRef MD) {
-  return wrap(MetadataAsValue::get(*unwrap(C), unwrap(MD)));
+LLVMValueRef LLVMMetadataAsValue(LLVMContextRef C, LLVMMetadataRef Metadata) {
+  auto *MD = unwrap(Metadata);
+  if (auto *VAM = dyn_cast<ValueAsMetadata>(MD))
+    return wrap(VAM->getValue());
+  else
+    return wrap(MetadataAsValue::get(*unwrap(C), MD));
```
Comment (nikic) [inline llvm/lib/IR/Core.cpp:1336]: Unrelated change?

**Review by nikic — COMMENTED** [review]
Can you please add a release note for the change?

File: llvm/lib/IR/Core.cpp (hunk 1 above)
Comment (maleadt) [reply llvm/lib/IR/Core.cpp]: Sorry, yeah, removed.

**Review by nikic — APPROVED** [review]
LGTM

**Comment by maleadt** [comment]
`TIMEOUT: lldb-api :: functionalities/fork/concurrent_vfork/TestConcurrentVFork.py` on linux-x64 unrelated
